File: vokimi_frontend/src/ts/page_classes/profile_edit_page/EditPageUserPagePrivacySectionData.ts
import type { PrivacyValues } from "../../enums/PrivacyValues";

export class EditPageUserPagePrivacySectionData {
    publishedTest: PrivacyValues;
    friends: PrivacyValues;
    followers: PrivacyValues;
    followings: PrivacyValues;
    constructor(publishedTest: PrivacyValues, friends: PrivacyValues, followers: PrivacyValues, followings: PrivacyValues) {
        this.publishedTest = publishedTest;
        this.friends = friends;
        this.followers = followers;
        this.followings = followings;
    }
}

File: vokimi_api/Src/dtos/responses/profile_editing_page/EditPageUserPagePrivacySectionData.cs
using vokimi_api.Src.db_related.db_entities.users;
using vokimi_api.Src.enums;

namespace vokimi_api.Src.dtos.responses.profile_editing_page
{
    public record class EditPageUserPagePrivacySectionData(
        PrivacyValues PublishedTest,
        PrivacyValues Friends,
        PrivacyValues Followers,
        PrivacyValues Followings
    )
    {
        public static EditPageUserPagePrivacySectionData FromUserPageSettings(
            UserPageSettings pageData
        ) => new(
            pageData.PrivacySettings.PublishedTests,
            pageData.PrivacySettings.Friends,
            pageData.PrivacySettings.Followers,
            pageData.PrivacySettings.Followings
        );
    }
}
